static/js/config.js

```diff
@@ -1,210 +1,152 @@
-    data["power-off"] = $("#input-power-off").val();
-  }
-
-  var formData = new FormData();
-  formData.append("section", "windows");
-  formData.append("config", JSON.stringify(data));
-  xmlHttp.send(formData);
+  await saveConfig("windows", data);
 }
 
-function saveModemConfig() {
-  var xmlHttp = new XMLHttpRequest();
-  xmlHttp.open("POST", "/api/config", true);
-  xmlHttp.setRequestHeader("Authorization", "Basic " + btoa("admin:feathers"));
-  xmlHttp.onload = async function () {
-    if (xmlHttp.status == 200) {
-      alert("Modem config saved");
-      loadConfig();
-    } else {
-      configError();
-    }
-  };
-  xmlHttp.onerror = async function () {
-    configError();
+async function saveModemConfig() {
+  const data = {
+    "initial-on-duration": document.querySelector("#input-initial-on-duration").value || undefined,
+    "find-modem-timeout": document.querySelector("#input-find-modem-timeout").value || undefined,
+    "connection-timeout": document.querySelector("#input-connection-timeout").value || undefined,
   };
 
-  var data = {};
-  if ($("#input-initial-on-duration").val() != "") {
-    data["initial-on-duration"] = $("#input-initial-on-duration").val();
-  }
-  if ($("#input-find-modem-timeout").val() != "") {
-    data["find-modem-timeout"] = $("#input-find-modem-timeout").val();
-  }
-  if ($("#input-connection-timeout").val() != "") {
-    data["connection-timeout"] = $("#input-connection-timeout").val();
-  }
+  await saveConfig("modemd", data);
+}
 
-  var formData = new FormData();
-  formData.append("section", "modemd");
-  formData.append("config", JSON.stringify(data));
-  xmlHttp.send(formData);
+async function saveThermalMotionConfig() {
+  const data = {
+    "do-tracking": document.querySelector("#input-do-tracking").checked,
+    "run-classifier": document.querySelector("#input-run-classifier").checked,
+    "tracking-events": document.querySelector("#input-tracking-events").checked,
+  };
+
+  await saveConfig("thermal-motion", data);
 }
 
-function saveThermalMotionConfig() {
-  var data = {};
-  data["do-tracking"] = $("#input-do-tracking").prop("checked");
-  data["run-classifier"] = $("#input-run-classifier").prop("checked");
-  data["tracking-events"] = $("#input-tracking-events").prop("checked");
+async function saveCommsConfig() {
+  const data = {
+    "enable": document.querySelector("#input-comms-enable").checked,
+    "trap-enabled-by-default": document.querySelector("#input-comms-trap-default").checked,
+    "comms-out": document.querySelector("#input-comms-type-select").value || undefined,
+    "bluetooth": document.querySelector("#input-comms-bluetooth").checked,
+    "power-output": document.querySelector("#input-comms-power-output").value || undefined,
+    "power-up-duration": document.querySelector("#input-comms-power-up-duration").value || undefined,
+    "trap-species": JSON.parse(document.querySelector("#input-comms-trap-species").value) || undefined,
```

Comment:
Should be fine, but careful of parsing JSON as it will throw instead of defaulting to undefined if it's invalid
```Javascript
try {
    protectSpecies = JSON.parse(document.querySelector("#input-comms-protect-species").value);
} catch (e) {
    console.error("Invalid JSON for Protect Species:", e);
    protectSpecies = undefined;
    alert("Please enter valid JSON for Protect Species.");
}
```